src/main/application/services/ui/ChangeUiVisibleUseCase.js

```diff
@@ -0,0 +1,41 @@
+import {DomainEventBus} from '../../../domain/service/DomainEventBus'
+import {inject} from '../../../core/ioc/ioc'
+import {DisplayStatusRepository} from '../../../domain/status/DisplayStatusRepository'
+import {DisplayStatus} from '../../../domain/status/DisplayStatus'
+import {EVENT_STATUS, EventStatus} from '../../../domain/status/EventStatus'
+
+export default class ChangeUiVisibleUseCase {
+  constructor({
+    domainEventBus = inject(DomainEventBus),
+    displayStatusRepository = inject(DisplayStatusRepository)
+  } = {}) {
+    this._domainEventBus = domainEventBus
+    this._displayStatusRepository = displayStatusRepository
+  }
+
+  execute({visible}) {
+    let eventStatus
+    if (visible === false) {
+      this._displayStatusRepository.setDisplayStatus({
+        newStatus: DisplayStatus.HIDDEN
+      })
+      eventStatus = EventStatus.USERACTIONCOMPLETE
+    } else {
+      this._displayStatusRepository.setDisplayStatus({
+        newStatus: DisplayStatus.VISIBLE
+      })
+      eventStatus = EventStatus.CMPUISHOWN
+    }
+    // TODO  get TCData
+    const TCData = {
```

Comment:
IMO, the EventStatusService should do the eventStatus work, maybe this should:

```
execute({visible}) {
  this._displayStatusRepository.setDisplayStatus({
    newStatus: visible ? DisplayStatus. VISIBLE : DisplayStatus.HIDDEN
   })
  this._eventStatusService.updateUiStatus() // emit the new status with the calc that it is already doing looking into the repositories and generating a new TCData
```

This would enable us to centralize the TCData generation for the event status in the EventStatusService only.
(the cmpLoaded will be needed to be emitted too)

Reply:
done.